File: merge_sort/JS/mergeSort.js
console.log("Merge Sort JS!");

const mergeSort = (array) => {

	// If array only has 1 element, no way to split.
	if(array.length < 2) return array;

	// Middle indext
	const midIndx = Math.floor(array.length/2);

	// Left side of the array
	const left = array.slice(0,midIndx);

	// Right side of the array
	const right = array.slice(midIndx);

	mergeSort(left);

	mergeSort(right);

	console.log(left, right);


}

mergeSort([1,2,3,4,5,6]);

File: merge_sort/Ruby/merge_sort.rb
p "Merge Sort Ruby!"

def merge_sort array
	# If array only has 1 element then no way to split.
	if array.length < 2
		return array;
	end

	mid_indx = (array.length/2).to_i;

	left = array[0...mid_indx]

	right = array[mid_indx...array.length]

	merge_sort left

 	merge_sort right

 	p left
 	p right

end

merge_sort [1, 2, 3, 4, 5, 6]